### PR TITLE
Add 1.4.*, 1.5.* and 1.1.6 tests for the ansible runner

### PR DESCRIPTION
**Pull request by brett060102**
add 1.4.* tests
add 1.5.* tests
add 1.1.6.runtime test
1.2.1 change task name to be .check to match other tests.

### CODE REVIEW

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.4.2/tasks/main.yml:12]: I would prefer to use return codes here.
The `correct` word is just a hardcoded value, that doesn't add any value.
In this case, this would be a correct return code, the next `else` would be an error one, and the 3rd one a correct one again

File: runner/ansible/vars/azure_env.yml
Comment (arbulu89) [inline runner/ansible/vars/azure_env.yml:26]: I would remove this `1.4.2` as it doesn't add any value, and we don't want the user changing this value, as it is custom predefined value

File: runner/ansible/vars/azure_env.yml
Comment (arbulu89) [inline runner/ansible/vars/azure_env.yml:27]: We don't need this `1.5.1` entry, we don't use it

File: runner/ansible/roles/1.5.2/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.5.2/tasks/main.yml:8]: Cannot we use the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/user_module.html module to test this?
Basically, we want to check if the `hacluster` user doesn't have `linux` as password, as this is a signal of a security hole

File: runner/ansible/roles/1.5.3/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.5.3/tasks/main.yml:11]: Do we need to have `sudo` here?
I thought ansible would execute this already with root rights

File: runner/ansible/vars/azure_env.yml
Comment (arbulu89) [inline runner/ansible/vars/azure_env.yml:28]: In fact, we don't need any of these `1.4.2`, `1.5.1`, `1.5.2` and `1.5.3` results here

File: runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml:9]: Cannot we do something to have all these `1.4.1` tests as an unique test?
This will make things really hard to work with, because they would be treated as individual tests, something we don't really like to do.
I would try to run all the commands here and compare them with the expected results or something.
It should be doable with shell,
Maybe, store the return code of the comparisons in an array, and check eventually if there is some incorrect value.

File: runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml:13]: This `3600` should we `expected[test_id]` I guess

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hi @brett060102 ,
My comments below.
In general, I don't like the approach of splitting the 1.4.1 test is subtests, it makes things harder to understand how to pack the result in the frontend. I think that we can put them all at one test.
Unless we are fine making them as independent tests, but we would need to change the identifier for something better (maybe adding a letter at the end). What do you think @stefanotorresi ?

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (aleksei-burlakov) [reply runner/ansible/roles/1.4.2/tasks/main.yml:12]: To my mind, it can as well be the "correct" string as long as it's consistent. We use however different string literals in the checks, so maybe, it'll be easier to implement with numeric codes.

File: runner/ansible/roles/1.5.2/tasks/main.yml
Comment (aleksei-burlakov) [reply runner/ansible/roles/1.5.2/tasks/main.yml:8]: > Cannot we use the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/user_module.html module to test this?
It looks to me like if we could, but let's wait for an answer from an ansible expert.

File: runner/ansible/roles/1.5.3/tasks/main.yml
Comment (aleksei-burlakov) [reply runner/ansible/roles/1.5.3/tasks/main.yml:11]: If it's all run as root, there is no need in sudo here of course.

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.4.2/tasks/main.yml:12]: I can go with a return code here.

File: runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.4.1.monitor_interval/tasks/main.yml:13]: As per our talk. I will cllaps all 1,4.1 tests into a single test with pass/fail return.

File: runner/ansible/roles/1.5.3/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.5.3/tasks/main.yml:11]: We cannot/should not depend on this being run as the root user.

File: runner/ansible/vars/azure_env.yml
Comment (brett060102) [reply runner/ansible/vars/azure_env.yml:27]: Correct, I should have pulled that line.

File: runner/ansible/vars/azure_env.yml
Comment (brett060102) [reply runner/ansible/vars/azure_env.yml:26]: OK. Will we work 1.4.2 to use exit status and fence_azure_arm not configured will be a pass.

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.4.2/tasks/main.yml:12]: Have moved to an RC for this one.

File: runner/ansible/roles/1.4.1/tasks/main.yml
Comment (diegoakechi) [inline runner/ansible/roles/1.4.1/tasks/main.yml:14]: I believe this `git ststus` was miss-typed here.

Comment (arbulu89) [reply runner/ansible/roles/1.4.1/tasks/main.yml:14]: Besides this typo, the check looks good though

File: runner/ansible/vars/azure_env.yml
Comment (arbulu89) [inline runner/ansible/vars/azure_env.yml:26]: We can remove this `1.4.2`. It is not used

**Review by arbulu89 — COMMENTED** [review]
Hi @brett060102 ,

The check `1.4.1` looks good now (except the type).
Let's remove the test `1.5.3` by now, so we can close this PR as soon as everything else is fixed

File: runner/ansible/roles/1.5.2/tasks/main.yml
Comment (arbulu89) [reply runner/ansible/roles/1.5.2/tasks/main.yml:8]: @brett060102 I have done some research and this option looks a bit nice:
(source: https://askubuntu.com/questions/611580/how-to-check-the-password-entered-is-a-valid-password-for-this-user)

```
salt=$(sudo getent shadow hacluster | cut -d$ -f3)
epassword=$(sudo getent shadow hacluster | cut -d: -f2)
match=$(python3 -c 'import crypt; print(crypt.crypt("linux", "$6$'${salt}'"))')
exit [ ${match} == ${epassword} ]
```

Could you give a try?
PD:
crypt is a native package in python3: https://docs.python.org/3.6/library/crypt.html

File: runner/ansible/roles/1.4.1/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.4.1/tasks/main.yml:14]: Thanks

File: runner/ansible/vars/azure_env.yml
Comment (brett060102) [reply runner/ansible/vars/azure_env.yml:26]: No, it is used now:
    [[ $(crm_attribute -t crm_config -G -n concurrent-fencing --quiet) = "{{ expected[test_id] }}" && exit 0

File: runner/ansible/roles/1.5.2/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.5.2/tasks/main.yml:8]: OK, I new get what you are trying to do. I will take a look.

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (arbulu89) [inline runner/ansible/roles/1.4.2/tasks/main.yml:10]: Is this working? I miss the closing brackets (`]]`)

File: runner/ansible/roles/1.5.2/tasks/main.yml
Comment (arbulu89) [reply runner/ansible/roles/1.5.2/tasks/main.yml:8]: Nice!

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hi @brett060102 ,
Good work.
We need to fix the `1.4.2` and the missing `]]`.
Once this is fixed we can merge

File: runner/ansible/roles/1.4.2/tasks/main.yml
Comment (brett060102) [reply runner/ansible/roles/1.4.2/tasks/main.yml]: fixed and really can't imagine how I toasted that one. should have stopped earlier.

**Review by arbulu89 — APPROVED** [review]
Thanks @brett060102 
Ready to go!

**Pull request merged**
